libsolidity/formal/SMTEncoder.cpp

```diff
@@ -3174,10 +3174,24 @@ vector<smtutil::Expression> SMTEncoder::symbolicArguments(FunctionCall const& _f
 		firstParam = 1;
 	}
 
-	solAssert((arguments.size() + firstParam) == functionParams.size(), "");
-	for (unsigned i = 0; i < arguments.size(); ++i)
-		args.push_back(expr(*arguments.at(i), functionParams.at(i + firstParam)->type()));
-
+	if (_funCall.isSuffixCall() && functionParams.size() == 2)
+	{
+		solAssert(!funType->hasBoundFirstArgument());
+		solAssert(arguments.size() == 1);
+		auto const* literalType = dynamic_cast<RationalNumberType const*>(arguments[0]->annotation().type);
```

Comment:
Let's actually go with the universal reference here:
```suggestion
		auto&& literalType = dynamic_cast<RationalNumberType const*>(arguments[0]->annotation().type);
```

I'm switching to that style in the main PR as well.